### PR TITLE
added return types to openmc.Settings

### PR DESCRIPTION
**Pull request by shimwell**
This PR aims to add return types to the ```openmc.Settings``` object.

This was originally proposed in PR #2016 along with moving the docstrings, however the docstring movement was not quite able to reproduce everything needed.

Now that #2016 is closed this aspect of #2016 has been split out so that is can be proposed separately  as discussed with @paulromano

### CODE REVIEW

File: openmc/settings.py [hunk 1]
```diff
@@ -298,191 +298,191 @@ def __init__(self):
...
         return self._inactive
 
     @property
-    def max_lost_particles(self):
+    def max_lost_particles(self) -> int:
         return self._max_lost_particles
 
     @property
-    def rel_max_lost_particles(self):
+    def rel_max_lost_particles(self) -> int:
```
Comment (paulromano) [inline openmc/settings.py:321]: Looks like this one is wrong here and elsewhere as well. Can you also fix the docstring above and the type annotation in the setter?
```suggestion
    def rel_max_lost_particles(self) -> float:
```

File: openmc/settings.py [hunk 2]
```diff
@@ -298,191 +298,191 @@ def __init__(self):
...
         return self._energy_mode
 
     @property
-    def max_order(self):
+    def max_order(self) -> int:
         return self._max_order
 
     @property
-    def source(self):
+    def source(self) -> Union[Source, typing.Iterable[Source]]:
```
Comment (paulromano) [inline openmc/settings.py:341]: Although the setter allows you to pass a scalar, it's always turned into a list
```suggestion
    def source(self) -> typing.List[Source]:
```

File: openmc/settings.py [hunk 3]
```diff
@@ -298,191 +298,191 @@ def __init__(self):
...
         return self._temperature
 
     @property
-    def trace(self):
+    def trace(self) -> typing.Iterable:
         return self._trace
 
     @property
-    def track(self):
+    def track(self) -> typing.Iterable[int]:
```
Comment (paulromano) [inline openmc/settings.py:425]:
```suggestion
    def track(self) -> typing.Iterable[typing.Iterable[int]]:
```

File: openmc/settings.py [hunk 4]
```diff
@@ -298,191 +298,191 @@ def __init__(self):
...
         return self._ufs_mesh
 
     @property
-    def resonance_scattering(self):
+    def resonance_scattering(self) -> dict:
         return self._resonance_scattering
 
     @property
-    def volume_calculations(self):
+    def volume_calculations(self) -> Union[VolumeCalculation, typing.Iterable[VolumeCalculation]]:
```
Comment (paulromano) [inline openmc/settings.py:441]:
```suggestion
    def volume_calculations(self) -> typing.List[VolumeCalculation]:
```

File: openmc/settings.py [hunk 5]
```diff
@@ -298,191 +298,191 @@ def __init__(self):
...
         return self._max_particles_in_flight
 
     @property
-    def write_initial_source(self):
+    def write_initial_source(self) -> bool:
         return self._write_initial_source
 
     @property
-    def weight_windows(self):
+    def weight_windows(self) -> Union[WeightWindows, typing.Iterable[WeightWindows]]:
```
Comment (paulromano) [inline openmc/settings.py:473]:
```suggestion
    def weight_windows(self) -> typing.List[WeightWindows]:
```

**Review by paulromano — CHANGES_REQUESTED** [review]

**Comment by shimwell** [comment]
Thanks for the review @paulromano , I've added your suggestions and fixed the rel_max_lost_particles elsewhere in the code (docstring and setter type).

**Review by paulromano — APPROVED** [review]

**Pull request merged**